### ISSUE TITLE
Docs page for bootstrap-salt (possibly impossible)

### ISSUE THREAD

**Issue by alan-cugler** [saltstack/salt#52278]
### Proposition
- Add bootstrap `README.md` from its repo and `--help` to the main salt docx
### Reasons
- It is strongly recommended to use the boostrap script when installing salt, because many of the OS packages are quickly out of date. 
- Though bootstrap is a separate repo, it is still a major step to get salt started. This warrants that it be integrated into the salt docx, so that the docx is a one stop shop for all your salt needs from beginning to end.
- We already have a [page](https://docs.saltstack.com/en/latest/topics/tutorials/salt_bootstrap.html) that simply pushes them to bootstrap-salt on github, so might as well place it on that same page.

I understand that we don't want bootstrap-salt integrated into salt (that wouldn't make any sense), but those docs would be helpful for beginners that are hunting docx.

So is there a way to integrate those two docx without the code?

**Comment by Ch3LL** [comment]
Previously the bootstrap page did actually have something similar to the README, but over time these two pages started to not mirror each other so we decided to only include a link to the README so we did not have wrong information in one place.

To integrate the README into our docs from the salt-bootstrap...i'm not aware of a way to do this with sphinx other then linking to it, unless someone on @saltstack/team-core has more sphinx foo and knows a better way to handle this.

**Comment by nixjdm** [comment]
I think we could add some code in the `doc/conf.py` that downloads the current bootstrap readme and then have that included in the build as normal. The python would download the file if it isn't present already, prior to building. The .gitignore should also be updated to ignore this readme. That way the website will always be up-to-date when built, assuming it's built from a fresh container or something (I don't know), but this readme largely won't bother developers.

There's two options I see for where to put the downloaded Readme - either in /ref somewhere as a `.rst`, or in `_static` as pre-rendered static content. If the later - which may be easier - it could be retrieved pretty easily from https://github.com/saltstack/salt-bootstrap with BeautifulSoup, for example.

Similarly, though it's a bit  more complex, some python included in / hooked into the conf.py could extract the `--help` content from the binary.

**Comment by gtmanfred** [comment]
I will say both of those options sound incredibly brittle, and if there is not a robust option that sphinx provides I don't think it is worth it.

**Comment by alan-cugler** [comment]
@gtmanfred sphinx doesn't have a feature that would cover this. 
This is outside the scope of what sphinx was designed for. 
However, this proposed idea is **not** outside the scope of salt docs.

Scraping probably sucks, so what about a script?
I would prefer to run a script with something like this as a first step when building the docs:
```BASH
#Getting the ReadMe
git clone git@github.com:saltstack/salt-bootstrap.git
cp salt-bootstrap/README.rst ./salt/docs/[place it should go]/README.rst
rm -rf salt-bootstrap/

#Getting the --help info
wget -O bootstrap-salt.sh https://bootstrap.saltstack.com
chmod +x bootstrap-salt.sh
./bootstrap-salt.sh -h >> ./salt/docs/[place it should go]/bootstrap_help.txt
rm bootstrap-salt.sh
``` 
This isn't perfect, but we get to avoid scraping and we get the added benefit of the text being in a `.rst` format for the ReadMe, and the `--help` will be extracted without any of the code from the bash script.

**Comment by alan-cugler** [comment]
Obviously, I am making some assumptions on how Saltstack has the salt docs built officially on their website.

**Comment by waynew** [comment]
FWIW, we *do* use a `make html` command (at least locally, I'm pretty sure we also do that to generate the site). As an alternative to using `git clone`, one *could* simply `curl` or `wget` the salt-bootstrap README.

**Comment by alan-cugler** [comment]
@waynew I'm cool with that, you want to approve this I will get it done.

**Comment by nixjdm** [comment]
Scraping the pre-rendered article out is surely brittle - agreed. Cloning / wget / requests, should only be as brittle as salt-bootstrap though, which being controlled by the same org, and largely stable these days, I wouldn't think should be a terrible liability. Not cloning, github's raw links ([example](https://raw.githubusercontent.com/saltstack/salt-bootstrap/develop/bootstrap-salt.sh)) are a veritable open api. Extracting the help menu should similarly not be all that brittle once you have the bootstrap shell script, imo. I think having the help menu is a big value add, however it gets in.

**Comment by waynew** [comment]
@alan-cugler Unless someone has some other information I'm not aware of, I'm 👍  on the idea.

I'm not sure if it would be worthwhile, but I *can* see the potential for adding maybe a `make html-offline` or something that basically looks like the existing command (or just `touch`-es the empty files where needed), in the case that someone is working on documentation somewhere without access to the 'net. Like an airplane without free wifi, for instance.

**Comment by alan-cugler** [comment]
@waynew assign me please

**Comment by gtmanfred** [comment]
No one can be assigned unless they are in the org.

**Comment by waynew** [comment]
@gtmanfred just tried it and I guess they can? 🤷‍♂️

**Comment by gtmanfred** [comment]
Oh, thats new then.

**Comment by alan-cugler** [comment]
please unassign me from this ticket

**Comment by nixjdm** [comment]
I'd still like to see this happen, one method or the other. Commenting to un-stale this.

**Comment by nixjdm** [comment]
@Ch3LL or @waynew Could you please add the [Feature](https://github.com/saltstack/salt/blob/master/.github/stale.yml#L18) label to prevent this staling? I don't mind if it continues to be "pending discussion". Though I don't have time atm to spearhead working this ticket, my interest in seeing it happen won't wane anytime soon :)

**Comment by alan-cugler** [comment]
I believe this issue will be solved via a possible project objective for the Docs working group.

**Comment by sagetherage** [comment]
@saltstack/docs-working-group
